### ISSUE TITLE
LookupMachineImage: Causes Invalid assembly manifest error

### ISSUE THREAD

**Issue by Console32** [aws/aws-cdk#21364]
### Describe the bug

I want to switch from a static AMI Map to a lookup. But when I use a Lookup Image, stack results in errors

current:
```
const image = MachineImage.genericLinux({
      [props.awsRegion]: props.ami,
    });
```

desired:
```
    const image = new LookupMachineImage({ name: 'whatever' } );
```

### Expected Behavior

cdk synth works

### Current Behavior

```
Error: Invalid assembly manifest:
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props requires property "domainName"
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props requires property "parameterName"
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props requires property "filter"
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props requires property "serviceName"
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props requires property "loadBalancerType"
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props requires property "loadBalancerType"
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props.account is not of a type(s) string
instance.missing[0].props requires property "aliasName"
instance.missing[0].props requires property "pluginName"
instance.missing[0].props is not any of <#/definitions/AmiContextQuery>,<#/definitions/AvailabilityZonesContextQuery>,<#/definitions/HostedZoneContextQuery>,<#/definitions/SSMParameterContextQuery>,<#/definitions/VpcContextQuery>,<#/definitions/EndpointServiceAvailabilityZonesContextQuery>,<#/definitions/LoadBalancerContextQuery>,<#/definitions/LoadBalancerListenerContextQuery>,<#/definitions/SecurityGroupContextQuery>,<#/definitions/KeyContextQuery>,<#/definitions/PluginContextQuery>

```

### Reproduction Steps

I am not able to reproduce this in a small example, can assist in pinning down the issue though. Just not sure where to start.

### CDK CLI Version

2.33.0 (build 859272d)

### Node.js Version

v16.15.1

### OS

Windows 11

### Language

Typescript

**Comment by Console32** [comment]
Changed Images to some prebuilt AMI's in a shared account and do now get the same error. Any clue where to start?

**Comment by rix0rrr** [comment]
Can you share relevant parts of `cdk.out/manifest.json` ?

**Comment by Console32** [comment]
What would be the relevant parts of the `manifest.json`. I can not see any obvious Problem in there.

**Comment by eerohakio** [comment]
Can confirm exact same issue happens when using
`route53.HostedZone.fromLookup`

Changing to fromHostedZoneAttributes seems to overcome the issue in this case.

**Comment by kochie** [comment]
I'm running into this issue as well.

I'm using a custom pipeline created with the `aws-cdk-lib/pipelines` package and it only happens when I update the `cdk.context.json` file with a `vpc.FromLookup` for when creating a new VPC. The interesting thing is it only seems to happen when trying to deploy a pipeline, the synth action works as intended when running without the pipeline.

https://gist.github.com/kochie/d5dc0e9cf653864ffd917d74f52d2031

**Comment by rix0rrr** [comment]
Please post a minimal reproducing example somewhere. Doing the following:

```ts
    new Instance(this, 'I', {
      machineImage: new LookupMachineImage({ name: 'whatever' } ),
      instanceType: new InstanceType('t2.micro'),
      vpc: new Vpc(this, 'Vpc'),
    });
```

For me does not reproduce the original report.

**Comment by Console32** [comment]
Hi @rix0rrr, 

I figured what is going on:
I am loading a configuration from a YAML file using a js function, the configuration contains an `awsAccountId` property.
The property ends up beeing of type number instead of string. This does not cause an error, until I try to look something up:

```
import { App, Stack, StackProps } from 'aws-cdk-lib';
import { MachineImage } from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

const config : any = {
  general: {
    awsAccount: 970942315402, // this caueses the error, if i change the account to a string (as it should be, everything works fine)
    awsRegion: 'eu-central-1',
  },
};

const app = new App();

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: { awsRegion: string } & StackProps) {
    super(scope, id, props);

    const image = MachineImage.lookup({ name: 'whatever' } );  // does not work
    // const image = MachineImage.genericLinux({ [props.awsRegion]: 'ami-1234567890' } ); // does work
    console.log(image.getImage(this));
  }
}

new MyStack(app, 'staging', {
  description: 'Staging stack',
  env: {
    account: config.general.awsAccount,
    region: config.general.awsRegion,
  },
  awsRegion: config.general.awsRegion,
});

app.synth();
```

So no bug from CDK, just some mistake of mine. 
Anyway - is there an option to catch this error early and post a different error message ?

**Comment by cschroed-usgs** [comment]
Good catch! Thanks for the workaround. I also experienced this while loading an account number from a yaml file, but in my case because I am running `Vpc.fromLookup`, CDK v1, and node 14.
```
$ npx cdk --version
1.126.0 (build f004e1a)
$ node --version
v14.20.0
```
I would also appreciate an earlier failure and a more descriptive error message.

**Comment by 89trillion-Miao** [comment]
> Hi @rix0rrr,
> 
> I figured what is going on: I am loading a configuration from a YAML file using a js function, the configuration contains an `awsAccountId` property. The property ends up beeing of type number instead of string. This does not cause an error, until I try to look something up:
> 
> ```ts
> import { App, Stack, StackProps } from 'aws-cdk-lib';
> import { MachineImage } from 'aws-cdk-lib/aws-ec2';
> import { Construct } from 'constructs';
> 
> const config : any = {
>   general: {
>     awsAccount: 970942315402, // this caueses the error, if i change the account to a string (as it should be, everything works fine)
>     awsRegion: 'eu-central-1',
>   },
> };
> 
> const app = new App();
> 
> export class MyStack extends Stack {
>   constructor(scope: Construct, id: string, props: { awsRegion: string } & StackProps) {
>     super(scope, id, props);
> 
>     const image = MachineImage.lookup({ name: 'whatever' } );  // does not work
>     // const image = MachineImage.genericLinux({ [props.awsRegion]: 'ami-1234567890' } ); // does work
>     console.log(image.getImage(this));
>   }
> }
> 
> new MyStack(app, 'staging', {
>   description: 'Staging stack',
>   env: {
>     account: config.general.awsAccount,
>     region: config.general.awsRegion,
>   },
>   awsRegion: config.general.awsRegion,
> });
> 
> app.synth();
> ```
> 
> So no bug from CDK, just some mistake of mine. Anyway - is there an option to catch this error early and post a different error message ?

Great! I changed accountId in yaml from integer to string (add double quote), then the same error disappear.
